Tests/NIOHTTP2Tests/SimpleClientServerTests.swift

```diff
@@ -1782,4 +1785,107 @@ class SimpleClientServerTests: XCTestCase {
+        let clientSettings = nioDefaultSettings + [HTTP2Setting(parameter: .maxFrameSize, value: maxFrameSize)]
+
+        // Begin by getting the connection up.
+        // We add the stream multiplexer here because it manages automatic flow control, and because we want the server
+        // to correctly confirm we don't receive a frame for the child stream.
+        let childHandler = FrameRecorderHandler()
+        try self.basicHTTP2Connection(clientSettings: clientSettings) { channel, streamID in
+            // Here we send a large response: 65535 bytes in size.
+            let responseHeaders = HPACKHeaders([(":status", "200"), ("content-length", "65535")])
+
+            var responseBody = self.clientChannel.allocator.buffer(capacity: 65535)
+            responseBody.writeBytes(Array(repeating: UInt8(0x04), count: 65535))
+
+            let respFrame = HTTP2Frame(streamID: streamID, payload: .headers(.init(headers: responseHeaders)))
+            channel.write(respFrame, promise: nil)
+
+            // Now prepare the large body.
+            let respBodyFrame = HTTP2Frame(streamID: streamID, payload: .data(.init(data: .byteBuffer(responseBody), endStream: true)))
+            channel.writeAndFlush(respBodyFrame, promise: nil)
+
+            return channel.pipeline.addHandler(childHandler)
+        }
+
+        // Now we're going to send a small request.
+        let headers = HPACKHeaders([(":path", "/"), (":method", "POST"), (":scheme", "https"), (":authority", "localhost")])
+
+        // We're going to open a stream and queue up the frames for that stream.
+        let handler = try self.clientChannel.pipeline.handler(type: HTTP2StreamMultiplexer.self).wait()
+        var reqFrame: HTTP2Frame? = nil
+
+        handler.createStreamChannel(promise: nil) { channel, streamID in
+            // We need END_STREAM set here, because that will force the stream to be closed on the response.
+            reqFrame = HTTP2Frame(streamID: streamID, payload: .headers(.init(headers: headers, endStream: true)))
+            channel.writeAndFlush(reqFrame, promise: nil)
+            return channel.eventLoop.makeSucceededFuture(())
+        }
+        (self.clientChannel.eventLoop as! EmbeddedEventLoop).run()
+
+        // Ok, we now want to send this data to the server.
+        self.interactInMemory(self.clientChannel, self.serverChannel)
+
+        // The server should have seen 1 window update frame for the connection only.
+        try self.serverChannel.assertReceivedFrame().assertWindowUpdateFrame(streamID: 0, windowIncrement: 65535)
+
+        // And only the request frame frame for the child stream, as there was no need to open its stream window.
+        childHandler.receivedFrames.assertFramesMatch([reqFrame!])
+
+        // No other frames should be emitted, though the client may have many.
+        self.serverChannel.assertNoFramesReceived()
+        XCTAssertNoThrow(try self.clientChannel.finish())
```

Comment:
```suggestion
        XCTAssertNoThrow(XCTAssertTrue(try self.clientChannel.finish().isClean))
```

Reply:
Here the client is expressly _not_ clean.